importMariaDBdmpFile/failover_import.py

```diff
@@ -0,0 +1,234 @@
+        # We do not run with '-q' to allow meaningful authentication error messages to be logged.
+        args = ['/usr/bin/scp',
+                '-o', 'PasswordAuthentication=no',
+                remoteUser + '@' + remoteHost + ':' + remotePath,
+                localDir
+                ]
+
+    runCommand(args)
+
+    logging.debug('remote fetch completed')
+
+    localPath = localDir + '/' + os.path.basename(remotePath)
+
+    suffix = os.path.splitext(localPath)[1]
+
+    if suffix == '.zip':
+        logging.debug('inflating fetched zip archive ...')
+        zip = zipfile.ZipFile(localPath, 'r')
+        zipContents = zip.namelist()
+        if len(zipContents) != 1:
+            raise Exception('Error: .zip archive must contain only 1 file.')
+        logging.debug('inflating to ' + localDir)
+        zip.extractall(localDir)
+        localPath = localDir + '/' + zip.namelist()[0]
+        logging.debug('inflated ' + localPath)
+        zip.close
+        logging.debug('zip archive inflation completed')
+
+    return localPath
+
+
+def importDB(optionsPath, importPath, retryCount):
+    '''
+    Use mysql to import the file
+    '''
+    # Tested using dump generated using the command
+    # > mysqldump --databases --lock-tables --dump-date \
+    #             --add-locks -p gocdb -r /tmp/dbdump.sql
+
+    args = ['/usr/bin/mysql',
+            '--defaults-extra-file=' + optionsPath,
+            '-e SOURCE ' + importPath
+            ]
+
+    count = 0
+
+    while count < retryCount:
+        count += 1
+        try:
+            logging.debug('loading database from dump file ...')
+            runCommand(args)
+            break
+        except Exception as exc:
+            logging.debug('mysql command import failed.')
+            if count < retryCount:
+                logging.error(str(sys.exc_info()[1]) + '. Retrying.')
+                snooze = min(0.1 * (pow(2, count) - 1), 20)
+                logging.debug('sleeping (' + '{:.2f}'.format(snooze) + 'secs)')
+                sleep(snooze)
+                logging.debug('retry ' + str(count) + ' of ' + str(retryCount))
+                # insert backoff time wait here
+                pass
+            else:
+                logging.debug('exceeded retry count for database load failures')
+                raise exc
+
+    logging.debug('database load completed')
+
+
+def archiveDump(importPath, archive, format):
+    '''
+    Save the dump file in the archive directory
+    '''
+
+    logging.debug('archiving dump file ...')
+
+    if not os.path.isdir(archive):
+        raise Exception('Archive directory ' + archive + ' does not exist.')
+
+    logging.debug('removing all .dmp files in ' + archive)
+
+    for oldDump in glob.iglob(archive + '/*.dmp'):
+        os.remove(oldDump)
+
+    archivePath = archive + '/' + strftime(format, gmtime()) + '.dmp'
+
+    logging.debug('moving ' + importPath + ' to ' + archivePath)
+
+    shutil.move(importPath, archivePath)
```

Comment:
```suggestion
    shutil.move(importPath, archivePath + '.dmp')
```

Reply:
Appended '.dmp' in the archivePath assignment for use in logging.debug stmt.